src/util/isEM.ts

```diff
@@ -1,8 +1,11 @@
 import { EM_TOKEN } from '../constants'
-import { Path } from '../types'
+import { Child } from '../types'
 
 /** Returns true if the Path is the EM_TOKEN. */
-export const isEM = (path: Path): boolean =>
-  path.length === 1 &&
-  !!path[0] &&
-  path[0].value === EM_TOKEN
+export const isEM = (thoughts: (string | Child)[]): boolean =>
+  thoughts.length === 1 &&
+  !!thoughts[0] &&
+  (
+    (thoughts[0] as Child).value === EM_TOKEN ||
+    (thoughts[0] as string) === EM_TOKEN
```

Comment:
This works on strings since the `value` property will return `undefined`, but it would be better to explicitly check `typeof thoughts[0]`. You could also use a function like this:

```js
/** Checks if an object is of type Path. */
const isPath = (o: Context | Path): o is Path =>
  o.length > 0 && Object.prototype.hasOwnProperty.call(o[0], 'value')
```

Reply:
Thanks!  I didn't know about user defined type guards.